cmd/dex/config.go

```diff
@@ -181,6 +182,20 @@ var storages = map[string]func() StorageConfig{
 	"mysql":      func() StorageConfig { return new(sql.MySQL) },
 }
 
+// isExpandEnvEnabled returns if os.ExpandEnv should be used for each storage and connector config.
+// Disabling this feature avoids surprises e.g. if the LDAP bind password contains a dollar character.
+// Returns true if the env variable "DEX_EXPAND_ENV" is unset or a truthy string, e.g. "true".
+// Returns false if the env variable is a falsy string, e.g. "false".
+// Returns an error if the env variable can't be parsed into a bool.
+func isExpandEnvEnabled() (bool, error) {
+	dexExpandEnv, exists := os.LookupEnv("DEX_EXPAND_ENV")
+	if !exists {
+		// Default, for downwards-compatibility: DEX_EXPAND_ENV = true
+		return true, nil
+	}
+	return strconv.ParseBool(dexExpandEnv)
```

Comment:
The reason I thought a negated variable name might be better is because variable expansion should be true by default, even if the content of the variable is considered invalid.

I agree it sounds bad, but I think we can keep `DEX_EXPAND_ENV`: I think we should handle the error returned by `ParseBool`: if there is an error, it should return true, otherwise return the parsed value. I think it's okay to silently fall back to true when the value of the variable is empty for example or some gibberish.

Reply:
Oh no, then I spent too much effort on the tests etc.... But you're right, silently using the default seems fine.